zephyr/lib/alloc.c

```diff
@@ -399,15 +382,12 @@ void *rmalloc(enum mem_zone zone, uint32_t flags, uint32_t caps, size_t bytes)
 #if CONFIG_L3_HEAP
 		heap = &l3_heap;
 		/* Uncached L3_HEAP should not be used */
-		if (!zone_is_cached(zone)) {
+		if (flags & SOF_MEM_FLAG_COHERENT) {
 			tr_err(&zephyr_tr, "L3_HEAP available for cached zones only!");
 			return NULL;
 		}
 		ptr = (__sparse_force void *)l3_heap_alloc_aligned(heap, 0, bytes);
 
-		if (!ptr && zone == SOF_MEM_ZONE_SYS)
-			k_panic();
```

Comment:
this can be problematic. Currently everywhere where `*_SYS` zones are used we omit return checking and rely on the system panicking. Arguably if we just drop this explicit panic we should anyway get one when trying to use the memory, but (1) it can happen later thus making debugging more difficult, (2) writing to `NULL` on ADSP doesn't cause exceptions #6093